src/strategies/implementations/list-tag-strategy.ts

```diff
@@ -0,0 +1,367 @@
+    } else {
+      // Handle opening ul tag
+      // Add line break before list container if needed
+      if (this.needsLineBreak(segments)) {
+        newSegments.push({
+          text: '\n',
+          ...currentStyle
+        });
+      }
+      
+      const newStyle = this.applyStyle(currentStyle, attributes, context.tagName);
+      
+      return {
+        newSegments,
+        updatedStyle: newStyle,
+        pushStyleToStack: true,
+        popFromStyleStack: false,
+        errors
+      };
+    }
+  }
+
+  public getTagNames(): string[] {
+    return [LIST_TAGS.UNORDERED_LIST];
+  }
+
+  public isLineBreak(): boolean {
+    return true;
+  }
+
+  public validateAttributes(attributes: string): { isValid: boolean; errors: string[] } {
+    return this.validateCommonListAttributes(attributes);
+  }
+
+  private validateCommonListAttributes(attributes: string): { isValid: boolean; errors: string[] } {
+    const errors: string[] = [];
+    
+    if (attributes.trim()) {
+      const validAttributePattern = /^(\s*(class|id|style|type|start)\s*=\s*["'][^"']*["']\s*)*$/i;
+      if (!validAttributePattern.test(attributes.trim())) {
+        errors.push('Invalid or unsupported attributes for list tag');
+      }
+    }
+    
+    return {
+      isValid: errors.length === 0,
+      errors
+    };
+  }
+}
+
+/**
+ * Strategy for ordered list containers: <ol>
+ * Handles opening/closing of ordered list tags
+ */
+class OrderedListStrategy extends BaseTagStrategy {
+  public applyStyle(currentStyle: TextStyle, attributes: string, tagName?: string): TextStyle {
+    // Stack management is handled by parent CompositeParseStrategy
+    return { ...currentStyle };
+  }
+
+  public parse(context: ParseContext): ParsedOutput {
+    const { isClosingTag, attributes, currentStyle, segments, remainingParts, currentIndex } = context;
+    const errors: string[] = [];
+    const newSegments: typeof context.segments = [];
+    
+    // Validate attributes
+    if (this.validateAttributes && !isClosingTag) {
+      const validation = this.validateAttributes(attributes);
+      if (!validation.isValid) {
+        errors.push(...validation.errors);
+      }
+    }
+    
+    if (isClosingTag) {
+      // Handle closing ol tag
+      // Stack management is handled by parent CompositeParseStrategy
+      
+      // Add line break after list if there's more content
+      if (this.hasMoreContent(remainingParts, currentIndex)) {
+        newSegments.push({
+          text: '\n',
+          ...currentStyle
+        });
+      }
+      
+      return {
+        newSegments,
+        updatedStyle: currentStyle,
+        pushStyleToStack: false,
+        popFromStyleStack: true,
+        errors
+      };
+    } else {
+      // Handle opening ol tag
+      // Add line break before list container if needed
+      if (this.needsLineBreak(segments)) {
+        newSegments.push({
+          text: '\n',
+          ...currentStyle
+        });
+      }
+      
```

Comment:
This logic seems co-related with above list strategy. could be refactored into utils methods for better readability